arbnode/inbox_tracker.go

```diff
@@ -462,6 +462,7 @@ func (t *InboxTracker) AddDelayedMessages(messages []*DelayedInboxMessage) error
 		}
 	}
 
+	firstPos := pos
```

Comment:
If I'm right, that will force a reorg of any batch added in any of the messages[] received.
But there is a possibility that some of the messages were already known and have not changed.
A few lines above we have:
```
if haveLastAcc == messages[len(messages)-1].AfterInboxAcc() {
			// We already have these delayed messages
			return nil
		}
```
I think that we want to test it for each message and use for firstPos the first one which it isn't true.

Reply:
Ah my bad, I read the loop above it and thought it did this but it only happens for snap sync apparently. I'll probably just modify the loop to always happen. Thanks for catching that.

Reply:
I've addressed this and updated the test to catch this issue.